### ISSUE TITLE
One possible error for fine tune mode in utils.py

### ISSUE THREAD

**Issue by rushingfox** [ViniciusMikuni/OmniLearned#16]
Within the restore_checkpoint function in utils.py, the final paragraph is:
```
    if optimizer is not None:
        try:
            optimizer.load_state_dict(checkpoint["optimizer"])
        except Exception:
            if is_main_node:
                print("Optimizer cannot be loaded back, skipping...")
```
However, loading the pre-train model's optimizer would give an error if a fine-tune regression task has different target dimension from the pre-train model: e.g. the pre-train model is trained by a multiple-labels regression, while the fine tune task is to regress only 1 target label.

The error can be avoided by changing the above code to:
```
    if optimizer is not None and not fine_tune:
        try:
            optimizer.load_state_dict(checkpoint["optimizer"])
        except Exception:
            if is_main_node:
                print("Optimizer cannot be loaded back, skipping...")
```
